pkg/router/template/plugin.go

```diff
@@ -303,13 +303,13 @@ func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, lookupSvc
 				if a.TargetRef != nil {
 					ep.TargetName = a.TargetRef.Name
 					if a.TargetRef.Kind == "Pod" {
-						ep.ID = fmt.Sprintf("pod:%s:%s:%s:%d", ep.TargetName, endpoints.Name, a.IP, p.Port)
+						ep.ID = fmt.Sprintf("pod:%s:%s:%s:%s:%d", ep.TargetName, endpoints.Name, p.Name, a.IP, p.Port)
 					} else {
-						ep.ID = fmt.Sprintf("ept:%s:%s:%d", endpoints.Name, a.IP, p.Port)
+						ep.ID = fmt.Sprintf("ept:%s:%s:%s:%d", endpoints.Name, p.Name, a.IP, p.Port)
 					}
 				} else {
-					ep.TargetName = ep.IP
-					ep.ID = fmt.Sprintf("ept:%s:%s:%d", endpoints.Name, a.IP, p.Port)
+					ep.TargetName = a.IP
```

Comment:
Ah, this must be the fuzz that inhibited automated cherry-picking.  Yeah, on release-4.6, `ep.IP` and `a.IP` may be different, and using `a.IP` is the correct thing to do whereas on release-4.5, `ep.IP` and `a.IP` should be the same, so it doesn't matter which you use.  ✓

Reply:
You are correct, this was one of the merge conflicts I had to resolve. There were also merge conflicts over `pkg/router/template/template_helper_test.go` since `TestClipHAProxyTimeoutValue` was recently backported.